core/ui.js

```diff
@@ -153,49 +250,59 @@ function addIconIfNeeded()
                                 \
                         <div id='incognito-option-presence-updates' class='incognito-options-item' style='cursor: pointer;'> \
                             <div class='checkbox-container " + UIClassNames.CHECKBOX_CONTAINER_CLASS + "' style='display:inline !important'> \
-                                    <div class='checkbox checkbox-incognito " + (options.presenceUpdatesHook ? "checked " + UIClassNames.RECTANGLE_CLASS + " "  + UIClassNames.CHECKBOX_CHECKED_CLASS + "'> <div class='checkmark " + UIClassNames.TICKED_CLASS + "'> </div>" : 
-                                    "unchecked " + UIClassNames.RECTANGLE_CLASS + "'> <div class='checkmark " + UIClassNames.UNTICKED_CLASS + "'> </div>") + "\
+                                    <div class='checkbox checkbox-incognito " + (options.presenceUpdatesHook ? "checked " + UIClassNames.RECTANGLE_CLASS + " " + UIClassNames.CHECKBOX_CHECKED_CLASS + "'> <div class='checkmark " + UIClassNames.TICKED_CLASS + "'> </div>" :
+                    "unchecked " + UIClassNames.RECTANGLE_CLASS + "'> <div class='checkmark " + UIClassNames.UNTICKED_CLASS + "'> </div>") + "\
                                 </div> \
                             </div> \
                             Don't send \"Last Seen\" updates \
                             <div class='incognito-options-description'>Blocks outgoing presence updates.</div> \
                         </div> \
+                        <div id='incognito-option-save-deleted-msgs' class='incognito-options-item' style='cursor: pointer;'> \
+                        <div class='checkbox-container " + UIClassNames.CHECKBOX_CONTAINER_CLASS + "' style='display:inline !important'> \
+                                <div class='checkbox checkbox-incognito " + (options.saveDeletedMsgs ? "checked " + UIClassNames.RECTANGLE_CLASS + " " + UIClassNames.CHECKBOX_CHECKED_CLASS + "'> <div class='checkmark " + UIClassNames.TICKED_CLASS + "'> </div>" :
+                    "unchecked " + UIClassNames.RECTANGLE_CLASS + "'> <div class='checkmark " + UIClassNames.UNTICKED_CLASS + "'> </div>") + "\
+                            </div> \
+                        </div> \
+                        Save Deleted Messages \
```

Comment:
Change the title capitalization to "Save deleted messages" and the subtitle to "Keeps deleted messages and shows the restored version" or something like that?

Reply:
edited ^^